components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/validators/LogoutRequestValidator.java

```diff
@@ -0,0 +1,184 @@
+            validator.accept(logoutRequest);
+            if (!samlMessageContext.getValidStatus()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Validate the SAML version of the logout request.
+     *
+     * @param logoutRequest {@link LogoutRequest} object to be validated.
+     * @throws SAMLLogoutException If SAML version validation fails.
+     */
+    private void isSAMLVersionValid(LogoutRequest logoutRequest) throws SAMLLogoutException {
+
+        if (!(SAMLVersion.VERSION_20.equals(logoutRequest.getVersion()))) {
+            String notification = "Invalid SAML Version in Logout Request. SAML Version should be equal to 2.0";
+            if (log.isDebugEnabled()) {
+                log.debug(notification);
+            }
+            samlMessageContext.setValidStatus(false);
+            String errorResponse = SAMLLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                    SSOConstants.StatusCodes.VERSION_MISMATCH, notification);
+            throw new SAMLLogoutException(notification, errorResponse, logoutRequest.getDestination(),
+                    samlMessageContext.getRelayState());
+        }
+    }
+
+    /**
+     * Validate the issuer of the logout request.
+     *
+     * @param logoutRequest {@link LogoutRequest} object to be validated.
+     * @throws SAMLLogoutException If Issuer of the Logout Request validation fails.
+     */
+    private void isIssuerValid(LogoutRequest logoutRequest) throws SAMLLogoutException {
+
+        if (logoutRequest.getIssuer().getValue() == null) {
+            String notification = "Issuer value cannot be null in the Logout Request";
+            if (log.isDebugEnabled()) {
+                log.debug(notification);
+            }
+            samlMessageContext.setValidStatus(false);
+            String errorResponse = SAMLLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                    SSOConstants.StatusCodes.REQUESTOR_ERROR, notification);
+            throw new SAMLLogoutException(notification, errorResponse, logoutRequest.getDestination(),
+                    samlMessageContext.getRelayState());
+        }
+        if (StringUtils.isBlank(logoutRequest.getIssuer().getFormat()) ||
+                !(ISSUER_FORMAT.equals(logoutRequest.getIssuer().getFormat()))) {
+            String notification = "Invalid Issuer Format in the logout request";
+            if (log.isDebugEnabled()) {
+                log.debug(notification);
+            }
+            samlMessageContext.setValidStatus(false);
+            String errorResponse = SAMLLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                    SSOConstants.StatusCodes.REQUESTOR_ERROR, notification);
+            throw new SAMLLogoutException(notification, errorResponse, logoutRequest.getDestination(),
+                    samlMessageContext.getRelayState());
```

Comment:
no need to both log and throw the exception here. The same log would be added twice. Since we are throwing the exception, I think we can do without the logging. WDYT?

Reply:
This exception is handled by framework side and this is used to send the saml error response to issuer.